src/jit/codegenxarch.cpp

```diff
@@ -4989,6 +4989,20 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
 #endif // defined(_TARGET_X86_)
 
+#ifdef FEATURE_AVX_SUPPORT
+    // When it's a PInvoke call and the call type is USER function, we issue VZEROUPPER here
+	// if the function contains 256bit AVX instructions, this is to avoid AVX-256 to Legacy SSE
+	// transition penalty, assume the user function may contains legacy SSE instruction
+    if (call->IsPInvoke() && call->gtCallType == CT_USER_FUNC && compiler->getFloatingPointInstructionSet() == InstructionSet_AVX)
+    {
+        if (getEmitter()->Contains256bitAVX())
+        {
```

Comment:
Another way to detect whether method has any 256-bit AVX instructions or 128/256-bit instructions, is to check whether any of gentree nodes are of SIMD vector types during lower register specification (lowerxarch.cpp:TreeNodeInfoInit()).

Contains256BitAvxInstr =  (getSIMDInstructionSet() == InstructionSet_AVX) and method has TYP_SIMD32 GenTree nodes as indicated by Lower register specification
ContainsAVXInstr = (getSIMDInstructionSet() == InstructionSet_AVX) and method has TYP_SIMD* GenTree nodes as indicated by Lower register specification.

This info could be used by codegen to emit vzeroupper as required.